style.md

```diff
@@ -1719,7 +1720,13 @@ BenchmarkGood-4  500000000   3.25 ns/op
 </td></tr>
 </tbody></table>
 
-### Prefer Specifying Map Capacity Hints
+### Prefer Specifying Container Capacity
+
+Specify container capacity where possible, as the compiler will allocate all
+required memory for that capacity up front rather than potentially allocating
+it in chunks as the container grows.
```

Comment:
The mention of the compiler seems a little too detailed here (since technically, most of the performance guidance affects what the compiler will do, though we don't call it out explicitly).

I also want to emphasize that it's not just that we allocate the memory in chunks (it might seem like instead of allocating 1mb, we'll allocate 5x200kb), but that we'll end up constantly resizing which increases total allocations, and also total time spent due to copies.

Perhaps:
"Specify container capacity where possible to allocate all required memory for the container up front. This avoids copies + resizing of the container as elements are added."

would replace mention of the compiler here (since it can happen at runtime for heap allocations), and the issue isn't just the chunks, but the resizing (e.g., overall more memory allocated + copies)

Reply:
Yeah, I think you're right. I generalized this a bit more to apply more correctly to both slices and maps (since map capacity hints don't guarantee that you'll "allocate all required memory for the container up front" obv) - not trying to be overly pedantic, just want the text to be accurate. Let me know what you think.